randovania/games/super_metroid/gui/preset_settings/super_patches_tab.py

```diff
@@ -0,0 +1,161 @@
+        ))
+        self.respin_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "respin"
+        ))
+        self.cant_use_supers_on_red_doors_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "cant_use_supers_on_red_doors"
+        ))
+        self.cheap_charge_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "cheap_charge"
+        ))
+        self.speedkeep_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "speedkeep"
+        ))
+        self.isj_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "infinite_space_jump"
+        ))
+        self.mb_rainbow_beam_nerf_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "nerfed_rainbow_beam"
+        ))
+        self.fix_spacetime_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "fix_spacetime"
+        ))
+        self.fix_heat_echoes_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "fix_heat_echoes"
+        ))
+        self.fix_screw_attack_menu_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "fix_screw_attack_menu"
+        ))
+        self.no_gt_checkbox.stateChanged.connect(functools.partial(
+            self._on_patch_checkbox_changed, "no_gt_code"
+        ))
+
+        self.vanilla_music_option.toggled.connect(functools.partial(
+            self._on_music_option_changed, MusicMode.VANILLA.value
+        ))
+        self.random_music_option.toggled.connect(functools.partial(
+            self._on_music_option_changed, MusicMode.RANDOMIZED.value
+        ))
+        self.no_music_option.toggled.connect(functools.partial(
+            self._on_music_option_changed, MusicMode.OFF.value
+        ))
+    @property
+    def uses_patches_tab(self) -> bool:
+        return True
+
+    def _on_patch_checkbox_changed(self, field_name: str, value: bool):
+        with self._editor as editor:
+            patch_configuration = editor.configuration.patches
+            new_config = dataclasses.replace(patch_configuration, **{field_name: value})
+            editor.set_configuration_field("patches", new_config)
+
+    def _on_music_option_changed(self, option: int, value: bool):
+        if value:
+            with self._editor as editor:
+                patch_configuration = editor.configuration.patches
+                new_config = dataclasses.replace(patch_configuration, **{"music": option})
+                editor.set_configuration_field("patches", new_config)
+
+    def on_preset_changed(self, preset: Preset):
+        patch_configuration = preset.configuration.patches
+
+        self.colorblind_checkbox.setChecked(patch_configuration.colorblind_mode)
```

Comment:
If you save that name -> checkbox dict into a variable, you can iterate over it here to also set the values. Something like:
```python
for name, checkbox in self.checkboxes.items():
    checkbox.setChecked(getattr(patch_configuration, name))
```